screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt

```diff
@@ -773,4 +821,69 @@ class EditTransactionViewModel @Inject constructor(
             save(closeScreen)
         }
     }
+
+    private suspend fun getAllTags(): ImmutableList<Tag> =
+        tagsRepository.findAll().toImmutableList()
+
+    private fun onTagSaved(name: String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            NotBlankTrimmedString.from(name.toLowerCaseLocal())
+                .onRight {
+                    val tag = with(tagMapper) { createNewTag(name = it) }
+                    tagsRepository.save(tag)
+                    this@EditTransactionViewModel.tags.value = getAllTags()
+                }
+
+            saveIfEditMode()
+        }
+    }
+
+    private fun associateTagToTransaction(selectedTag: Tag) {
+        viewModelScope.launch(Dispatchers.IO) {
+            val associatedId = AssociationId(loadedTransaction().id)
+            tagsRepository.associateTagToEntity(associatedId, selectedTag.id)
+            transactionAssociatedTags.value =
+                tagsRepository.findByAssociatedId(associatedId).toImmutableList()
+        }
+    }
+
+    private fun removeTagAssociation(selectedTag: Tag) {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
No need to switch to IO thread, if the repository is implemented correctly - it must be main-safe and does the `withContext(dispatcherProvider.io)` we use the dispatcher provider so stuff can be unit tested w/o being flaky

Reply:
I did this as I didn't want anything to happen on the main thread not even assigning simple variable.
The app is not optimized to didn't want to worsen it. 

Yes I have used withContext(dispatcherProvider.io) on repository.

Reply:
Actually, assigning simply variables outside the main thread doesn't improve performance but increases complexity. One can argue that it even worsens it cuz it consufes Compose and makes recompositions more complicated. 

These micro optimizations doesn't help. The problem is in our algorithms which are terrible... See #2990

Reply:
Let's remove it in future PRs cuz other contributors will start to copy it

Reply:
Yes but we are performing some business logic 
eg. converting list to immutable list running this on main thread will add delay with screen refresh rates now being 90hz 
until algorithms are fixed I feel we should off load such tasks to another thread

Reply:
Okay, does it make more sense to move it to `dispatcherProvider.default` so we don't congest the IO thread pool?